ctms/app.py

```diff
@@ -1122,5 +1124,41 @@ def stripe_pubsub(
     return {"status": "OK", "count": count}
 
 
+@app.delete(
+    "/ctms/gdpr/{email}",
+    tags=["Public"],
+    summary="delete email from database",
+    responses={
+        404: {"model": NotFoundResponse},
+    },
+    response_model=GDPRDeleteResponse,
+)
+def gdpr_delete_email(
+    email: str,
+    db: Session = Depends(get_db),
+    api_client: ApiClientSchema = Depends(get_enabled_api_client),
+):
+    email_to_find = email.lower()
+    ids = all_ids(primary_email=email_to_find)
+    contacts = get_contacts_by_ids(db, **ids)
+
+    if not contacts:
+        raise HTTPException(status_code=404, detail=f"email {email} not found!")
+
+    dropped = list()
+
+    for contact in contacts:
+        gdpr_delete(db=db, email_id=contact.email.email_id)
+
+        dropped.append(
+            {
+                "primary_email": contact.email.primary_email,
+                "email_id": str(contact.email.email_id),
+            }
+        )
+
+    return GDPRDeleteResponse(status="ok", dropped=dropped)
```

Comment:
We could return the whole contact info like in the `/identities` endpoints.
```py
return [contact.as_identity_response() for contact in contacts]
```

Reply:
makes way more sense to me, dropped the GDPRDeleteResponse :+1: